addons/mail/static/src/model/model_manager.js

```diff
@@ -385,6 +393,27 @@ export class ModelManager {
      */
     _applyModelDefinition(model) {
         const definition = registry.get(model.name);
+        if (definition.get('template')) {
+            const ModelComponent = { [model.name]: class extends Component {} }[model.name];
+            if (definition.get('componentSetup')) {
+                Object.assign(ModelComponent.prototype, { setup() {
+                    return definition.get('componentSetup').call(this);
+                } });
+            }
+            Object.defineProperty(
+                ModelComponent.prototype,
+                definition.get('templateGetter') ? definition.get('templateGetter') : 'record',
+                { get() {
+                    return this.props.record;
+                } },
+            );
+            Object.assign(ModelComponent, {
+                props: { record: Object },
+                template: definition.get('template'),
+            });
+            registerMessagingComponent(ModelComponent);
+            model.hasComponent = true;
+        }
```

Comment:
Creating the component in the ModelManager might be a problem since that means you can no longer retrieve messaging components before the ModelManager starts. Typically, components that depend on some container, which itself makes use of `getMessagingComponent`, cannot be converted.

Reply:
The JS of these components still exist for that reason. We will tackle them in follow PR, e.g. by defining a generic container layer that properly handles awaiting availability of registered messaging component for mount.